### PR TITLE
Show disabled entity names on the device page

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Show the name for disabled entities on the device info page.

Implementation feels _a little_ hacky. We only get the original name when we fetch an individual entry, so the `<device-entities-card>` will fetch all individual entries of disabled entities when the user clicks on "Show disabled entities". It will fetch max 3 in parallel.

image

Idea by @elupus (if I recall correctly)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by elupus** [comment]
Looks nice from a visual standpoint. But feels we should be able to get names at the same time we get the list of entity IDs? To avoid the havkyness.

**Comment by balloob** [comment]
The entity registry entities are fetched 2-3 components up and are the non-extended ones. We need the extended ones which can only be fetched 1 at a time right now.

File: src/panels/config/devices/device-detail/ha-device-entities-card.ts [hunk 1]
```diff
@@ -115,6 +130,28 @@ export class HaDeviceEntitiesCard extends LitElement {
...
+
+    const worker = async () => {
+      if (toFetch.length === 0) {
+        return;
+      }
+
+      const entityId = toFetch.pop()!.entity_id;
+      const entry = await getExtendedEntityRegistryEntry(this.hass, entityId);
+      this._extDisabledEntityEntries![entityId] = entry;
+      this.requestUpdate();
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/ha-device-entities-card.ts:147]:
```suggestion
      this.requestUpdate("_extDisabledEntityEntries");
```

File: src/panels/config/devices/device-detail/ha-device-entities-card.ts [hunk 2]
```diff
@@ -115,6 +130,28 @@ export class HaDeviceEntitiesCard extends LitElement {
 
   private _toggleShowDisabled() {
     this.showDisabled = !this.showDisabled;
+    if (!this.showDisabled || this._extDisabledEntityEntries !== undefined) {
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/ha-device-entities-card.ts:133]: probably not something we will run into now, but if `entities` is changed we should also refetch `_extDisabledEntityEntries`

**Pull request merged**